### DISCUSSION TITLE
Question: Is GCNConv only available for undirected graphs?

### DISCUSSION THREAD

**Discussion by Struggle-Forever** [pyg-team/pytorch_geometric#4954] [General]
GCNConv()

**Comment by rusty1s** [comment]
Yes, it should only be used for undirected graphs. If you want to apply it to directed graphs, I suggest you apply two `GCNConv` layers and sum up their outputs:
```python
out = conv1(x, edge_index) + conv2(x, edge_index.flip([0]))
```

**Reply by Struggle-Forever** [reply]
Can I understand it like this.t like this： conv1(x, edge_index) is an undirected operation for graph encoders.  conv2(x, edge_index.flip([0]))  do directed information propagation via edge_index.flip([0])? What puzzles me is that the theoretical derivation of GCN is based on undirected graphs, is this feasible?

**Reply by rusty1s** [reply]
Yes, this is true. As you correctly mention, you will throw the theoretical derivation of GCN that way - my assumption is that it still works quite well. Alternatively, you can replace your `GCNConv` with `SAGEConv`, or have a look at [this paper](https://arxiv.org/pdf/2004.13970.pdf).

**Reply by Struggle-Forever** [reply]
thanks  i finally adopt GAT,  and the GraphSAGE is a good choice.
